### PR TITLE
Add support for persistent keys

### PR DESCRIPTION
**Pull request by magnouvean**
Should address the persistence we have discussed in #120. To allow for the functionality I had to do quite a bit of refactoring, and therefore this is quite a big change moving some of the nix code to the python script. Therefore there could also be some bugs introduced (though none I have found just yet). I very much welcome anyone to test on their own configs and let me know how it works before we merge (it's probably wise to backup the config-files just in case some bug I have overseen). This also hopefully should open up for solving #119 more easily as we can just enable persistence for some of the keys (granted we can find out which ones are at fault).

### CODE REVIEW

File: lib/types.nix [hunk 1]
```diff
@@ -27,6 +27,14 @@ let
           adding [$e] to the end of the key.
         '';
       };
+      persistent = lib.mkOption {
```
Comment (Asqiir) [inline lib/types.nix:30]: Does this allow the user to set `persistent = true;` and `immutable = true;` at the same time? What is the desired behaviour of plasma-manager in that case?
(Imo this shouldn't be possible.)

Comment (Asqiir) [inline lib/types.nix:30]: And what about 
```nix
somekey = {
  value = "myval";
  persistent = "true";
};
```
Which value should be set on activation?

Comment (magnouvean) [reply lib/types.nix:30]: I agree, it shouldn't be possible. It's an easy fix luckily

Comment (magnouvean) [reply lib/types.nix:30]: As of the latest version I am pretty sure this isn't possible to do. It won't allow values when persistence is enabled as it doesn't really make sense.

Comment (Asqiir) [reply lib/types.nix:30]: that's great to hear :)

question:
doesn't that mean there are only 4 options left(?):
1. no value -> persistent by default
2. immutable and a value
3. immutable without a value (should that be possible?)
4. persistent and no value, explicitly

if option 3 should not be possible we could do something like: (just a suggestion, maybe for later)

```nix
somekey = null; # means persistent, is the default
someotherkey = "value"; # makes it immutable
```

then we do not need the immutable/persistent/value flags anymore (?)

Comment (magnouvean) [reply lib/types.nix:30]: 3 shouldn't have been possible, but there wasn't any checks in the code which made it illegal (but just before you commented I added a check for this as well :)). There also is shell-expansion of variables, which pretty much is the same as immutability. I also think it is possible to the explicitly setting the key to `null` which should remove the key from the file altogether. There's plenty of options and combinations, but I think we have gotten rid of most of the combinations that don't make sense.

**Comment by magnouvean** [comment]
Did the functionality work the way you expected @Asqiir? In the current state it works for my needs anyway, so if there isn't anything I have missed I think we can merge this quite soon.

**Comment by Asqiir** [comment]
It does!
And I see now that leaving room for more flags makes sense :)
Thanks for building this

**Pull request merged**